usr/share/rear/layout/save/GNU/Linux/230_filesystem_layout.sh

```diff
@@ -111,10 +111,39 @@ fi
             # but ensure docker_root_dir is not empty (otherwise any mountpoint string matches "^" which
             # would skip all mountpoints), see https://github.com/rear/rear/issues/1989#issuecomment-456054278
             if test "$docker_root_dir" ; then
-                if echo "$mountpoint" | grep -q "^$docker_root_dir" ; then
+                if echo "$device" | grep -q "^${docker_root_dir}/" ; then
```

Comment:
@gdha 
I am still confused here because the comment reads
```
# If docker_root_dir is the beginning of the mountpoint string then the filesystem is under Docker control
```
but the actual test is not the mountpoint string but the device string
```
if echo "$device" | grep -q "^${docker_root_dir}/" ; then
```
so I guess the comment is wrong but I still do not understand
what value is in $docker_root_dir and $mountpoint versus $device

My offhanded expectation would be that $device is something like /dev/sdb2
and its mountpoint is something like /path/to/some/mountpoint/directory
and the Docker root dir is something like /path/to/docker/stuff
and therein are the mountpoint dirrectories for Docker like /path/to/Docker/stuff/mounts
so that Docker's things would have a mountpoint string with leading Docker root dir
i.e. same locic as for the subsequent Longhorn case where the test is
```
if echo "$mountpoint" | grep -q "^/dev/longhorn/pvc-" ; then
```
so some `longhorn` specific thingy (a device node in this case)
is leading part of the mountpoint of what is mounted for Longhorn.

Reply:
@jsmeix Good that you were confused! That is the result when you're busy with too many items at the same time. `docker_root_dir` is a mount point so the if should be against mountpoint and not device. The longhorn is correct - check for the device.